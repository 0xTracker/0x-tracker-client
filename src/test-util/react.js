import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
