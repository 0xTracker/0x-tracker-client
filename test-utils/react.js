import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

const withRedux = (
  component,
  { initialState, store = createStore(reducer, initialState) } = {},
) => ({
  component: <Provider store={store}>{component}</Provider>,
  store,
});

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

function renderWithRedux(component, options) {
  const wrappedComponent = withRedux(component, options);

  return {
    ...render(wrappedComponent.component),
    store: wrappedComponent.store,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRedux, renderWithRouter, withRedux };
