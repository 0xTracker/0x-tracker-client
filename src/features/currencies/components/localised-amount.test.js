import React from 'react';

import { renderWithAppContext } from '../../../../test-utils/react';
import LocalisedAmount from './localised-amount';

it('should render USD amount without conversion rates loaded', () => {
  const { container } = renderWithAppContext(
    <LocalisedAmount amount={500.2} />,
  );

  expect(container).toMatchSnapshot();
});

it('should render CNY amount when conversion rate is known', () => {
  const { container } = renderWithAppContext(
    <LocalisedAmount amount={500.2} />,
    {
      initialState: { preferences: { currency: 'CNY' }, rates: { CNY: 6.86 } },
    },
  );

  expect(container).toMatchSnapshot();
});

it('should render loading indicator when display currency is CNY and conversion rate is unknown', () => {
  const { container } = renderWithAppContext(
    <LocalisedAmount amount={500.2} />,
    {
      initialState: { preferences: { currency: 'CNY' } },
    },
  );

  expect(container).toMatchSnapshot();
});
