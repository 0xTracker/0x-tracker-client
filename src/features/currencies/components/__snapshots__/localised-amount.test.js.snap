// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CNY amount when conversion rate is known 1`] = `
<div>
  Â¥3,431.37
</div>
`;

exports[`should render USD amount without conversion rates loaded 1`] = `
<div>
  $500.20
</div>
`;

exports[`should render loading indicator when display currency is CNY and conversion rate is unknown 1`] = `
<div>
  <div
    style="fill: #333; height: 22px; width: 22px;"
  >
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M0 12 V20 H4 V12z"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      
  
      <path
        d="M0 12 V20 H4 V12z"
        opacity="0.5"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0.1s"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      
  
      <path
        d="M0 12 V20 H4 V12z"
        opacity="0.25"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0.2s"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</div>
`;
